### DISCUSSION TITLE
order list by relation field

### DISCUSSION THREAD

**Discussion by soluzione-software** [Laravel-Backpack/community-forum#945] [Q&A (Help)] [answered]
I have a table sales and a relation customer with customers table
in my customers table I have business_name field
I want to order my sales list by this field that is in relation
this is my addColumn:
$this->crud->addColumn([
            'label' => __('business_name'), // Table column heading
            'type' => "select",
            'name' => 'customer_id', // the column that contains the ID of that connected entity;
            'entity' => 'customer', // the method that defines the relationship in your Model
            'attribute' => "business_name", // foreign key attribute that is shown to user
            'model' => Customer::class,
            'orderable'  => true,
        ]);
but order doesn't work, why?
thanks

**Comment by karandatwani92** [comment, marked as answer]
Hey @soluzione-software 

The attribute `'orderable' => true,` is used only when you want to set [Custom Order Logic for Column](https://backpackforlaravel.com/docs/6.x/crud-columns#custom-order-logic-for-columns).

I'm also using a select column in one of my projects, and it is orderable by default. Here's the definition of it:
```php
// Column inside setupListOperation()
CRUD::column('category_id')->type('select')->label('Category')->entity('category')->attribute('name')
->model(App\Models\VisitorCategory::class);
```

```php
// Model relationship
public function category()
{
   return $this->belongsTo('App\Models\VisitorCategory', 'category_id');
}
```

I hope this example will help you.

**Reply by soluzione-software** [reply]
thanks for the reply
i made this
$this->crud->addColumn([
            'name' => 'customer_id', // the column that contains the ID of that connected entity;
            'type' => "select",
            'label' => __('business_name'), // Table column heading
            'entity' => 'customer', // the method that defines the relationship in your Model
            'attribute' => "business_name", // foreign key attribute that is shown to user
            'model' => Customer::class,
        ]);

in my model Sale:
    public function customer(){
        return $this->belongsTo(Customer::class);
    }

but watching data the order is by customer_id and not by business_name
why?
thanks very much

**Reply by karandatwani92** [reply]
OK, I get you now; you need to set [Custom Order Logic for Column](https://backpackforlaravel.com/docs/6.x/crud-columns#custom-order-logic-for-columns).
